homeassistant/components/abode/config_flow.py

```diff
@@ -25,53 +26,146 @@ def __init__(self):
             vol.Required(CONF_PASSWORD): str,
         }
+        self.mfa_data_schema = {
+            vol.Required(CONF_MFA): str,
+        }
+
+        self._cache = None
+        self._mfa_code = None
+        self._password = None
+        self._polling = False
+        self._username = None
+
+    async def _async_abode_login(self, step_id):
+        """Handle login with Abode."""
+        self._cache = self.hass.config.path(DEFAULT_CACHEDB)
+        errors = {}
+
+        try:
+            await self.hass.async_add_executor_job(
+                Abode, self._username, self._password, True, False, False, self._cache
+            )
+
+        except (AbodeException, ConnectTimeout, HTTPError) as ex:
+            if ex.errcode == MFA_CODE_REQUIRED[0]:
+                return await self.async_step_mfa()
+
+            LOGGER.error("Unable to connect to Abode: %s", ex)
+
+            if ex.errcode == HTTP_BAD_REQUEST:
+                errors = {"base": "invalid_auth"}
+
+            else:
+                errors = {"base": "cannot_connect"}
+
+        if errors:
+            return self.async_show_form(
+                step_id=step_id, data_schema=vol.Schema(self.data_schema), errors=errors
+            )
+
+        return await self._async_create_entry()
+
+    async def _async_abode_mfa_login(self):
+        """Handle multi-factor authentication (MFA) login with Abode."""
+        try:
+            # Create instance to access login method for passing MFA code
+            abode = Abode(
+                auto_login=False,
+                get_devices=False,
+                get_automations=False,
+                cache_path=self._cache,
+            )
+            await self.hass.async_add_executor_job(
+                abode.login, self._username, self._password, self._mfa_code
+            )
+
+        except AbodeAuthenticationException:
+            return self.async_show_form(
+                step_id="mfa",
+                data_schema=vol.Schema(self.mfa_data_schema),
+                errors={"base": "invalid_mfa_code"},
+            )
+
+        return await self._async_create_entry()
+
+    async def _async_create_entry(self):
+        """Create the config entry."""
+        config_data = {
+            CONF_USERNAME: self._username,
+            CONF_PASSWORD: self._password,
+            CONF_POLLING: self._polling,
+        }
+        existing_entry = await self.async_set_unique_id(self._username)
+
+        if existing_entry:
+            self.hass.config_entries.async_update_entry(
+                existing_entry, data=config_data
+            )
+            # Reload the Abode config entry otherwise devices will remain unavailable
+            self.hass.async_create_task(
+                self.hass.config_entries.async_reload(existing_entry.entry_id)
+            )
+
+            return self.async_abort(reason="reauth_successful")
+
+        return self.async_create_entry(title=self._username, data=config_data)
```

Comment:
@MartinHjelmare I'm assuming what you're mentioning in the other code comment I made is in reference to this block of code? As you mentioned and looking at `_abort_if_unique_id_configured` in the base `ConfigFlow` class, it looks like this is what `_abort_if_unique_id_configured` is doing but I just realized for users that have an existing Abode config flow, I don't think I was setting the `unique_id` property before so they're likely just set to `None`. I'm still looking into this but if that's the case, I'm guessing I'll need to add a check to update the users config flow with a `unique_id` if it doesn't already exist.

Reply:
I this this part is ok for now.

You're correct that we should update the existing entries and make sure they have unique id set.